File: rules/go/tests/go.cmd-injection/cmd.go
package main

import (
	"net/http"
	"os/exec"
)

func direct(r *http.Request) {
	exec.Command("sh", "-c", "ping "+r.FormValue("host")) // crit:expect go.cmd-injection
}

func viaVar(r *http.Request) {
	host := r.URL.Query().Get("host")
	// crit:expect go.cmd-injection
	exec.Command("sh", "-c", "ping "+host)
}

func fixed() {
	exec.Command("ls", "-la") // crit:expect-not go.cmd-injection
}

func internal(host string) {
	exec.Command("ping", host) // crit:expect-not go.cmd-injection
}

File: rules/rust/tests/rs.cmd-injection/cmd.rs
use std::process::Command;

fn direct() {
    Command::new(std::env::args().nth(1).unwrap()); // crit:expect rs.cmd-injection
}

fn via_var() {
    let prog = std::env::var("PROG").unwrap();
    // crit:expect rs.cmd-injection
    Command::new(prog);
}

fn fixed() {
    Command::new("ls"); // crit:expect-not rs.cmd-injection
}

fn internal(prog: String) {
    Command::new(prog); // crit:expect-not rs.cmd-injection
}
